docs(linter): Use backticks for code elements across more rule diagnostics

connorshea opened the pull request:
Improves the consistency of diagnostic formatting in 30ish rules, and improves the diagnostic messages in editors that render markdown, plus makes it clearer which parts refer to code for end-users, even with the raw text.

-lso adjust the phrasing of some diagnostics slightly.

camc314 commented:
### Merge activity

* **Nov 22, 11:14 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 22, 11:14 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
docs(linter): Use backticks for code elements across more rule diagnostics (#15958)

Improves the consistency of diagnostic formatting in 30ish rules, and improves the diagnostic messages in editors that render markdown, plus makes it clearer which parts refer to code for end-users, even with the raw text.

-lso adjust the phrasing of some diagnostics slightly.
